### PR TITLE
DOCSP-43071: Bulk operations

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43071
Staging - https://deploy-preview-54--docs-c.netlify.app/write/bulk-write/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/write/bulk-write.txt [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+Additional Information
+----------------------
+
+To learn how to perform individual write operations, see the following guides:
+
+- :ref:`c-write-insert`
+- :ref:`c-write-update`
+- :ref:`c-write-delete`
+- :ref:`c-write-replace`
```
Comment (norareidy) [inline source/write/bulk-write.txt:282]: Note to reviewer: these links will be broken until #52 and #53 are merged

File: source/write/update.txt [hunk 2]
```diff
@@ -57,6 +57,8 @@ Each update function accepts the following parameters:
   and their usage, see the :manual:`Field Update Operators guide
   </reference/operator/update-field/>` in the {+mdb-server+} manual.
 
+- Options document: Specifies options to customize the operation, or ``NULL``.
```
Comment (stephmarie17) [inline source/write/update.txt:60]: [s] This list isn't fully from the diff, but I think the [guidance around bolding terms](https://www.mongodb.com/docs/meta/style-guide/style/lists/#list-that-provides-terms-and-more-information) would apply her as well.

**Review by stephmarie17 — APPROVED** [review]
LGTM with a few suggestions!

File: source/includes/write/bulk-write.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        // Creates insert operation instructions and adds the operation to the bulk write
+        // start-bulk-insert-one
+        bson_t *insert_doc = BCON_NEW (
+            "name", BCON_UTF8 ("Mongo's Deli"),
+            "cuisine", BCON_UTF8 ("Sandwiches"),
+            "borough", BCON_UTF8 ("Manhattan"),
+            "restaurant_id", BCON_UTF8 ("1234")
+        );
+
+        mongoc_bulk_operation_insert_with_opts (bulk, insert_doc, NULL);
```
Comment (kevinAlbs) [inline source/includes/write/bulk-write.c:30]: Add an out-parameter for a `bson_error_t`. Suggest checking the return:

```c
bson_error_t error;
if (!mongoc_bulk_operation_insert_with_opts (bulk, insert_doc, NULL, &error)) {
    fprintf (stderr, "error adding insert operation: %s\n", error.message);
}
```

Suggestion also applies to other `mongoc_bulk_operation_*_with_opts` calls.

File: source/write/bulk-write.txt [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+function:
+
+.. literalinclude:: /includes/write/bulk-write.c
+   :start-after: start-bulk-run
+   :end-before: end-bulk-run
+   :language: c
+   :dedent:
+
+If any of the write operations fail, the {+driver-short+} raises a
+``mongoc_bulkwriteexception_t`` error and does not perform any further operations.
```
Comment (kevinAlbs) [inline source/write/bulk-write.txt:213]: Remove mention of `mongoc_bulkwriteexception_t`. `mongoc_bulkwriteexception_t` applies to the [mongoc_bulkwrite_t](https://mongoc.org/libmongoc/current/mongoc_bulkwrite_t.html). `mongoc_bulkwrite_t` requires server 8.0. `mongoc_bulk_operation_t` can be used with any supported server version.

```suggestion
If any of the write operations fail, the {+driver-short+} sets the output error and does not perform any further operations.
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with one fix.

**Pull request merged**